### PR TITLE
Add templates for deploying web app on Kubernetes

### PR DESCRIPTION
**Pull request by scottslowe**
Add templates in C#, Go, Python, TypeScript, and YAML for deploying a web application on Kubernetes. Creates a ConfigMap, a Deployment that mounts the ConfigMap, and a Service that exposes the Pods managed by the Deployment.

### CODE REVIEW

**Comment by scottslowe** [comment]
@cnunciato If you have a chance to test the C# template, that would be great. The laptop I'm traveling with doesn't have .NET installed, and standing up a temporary EC2 instance is...painful when it comes to EKS clusters (which is what I've been using to test the templates). The rest of the templates should work without any issues.

File: webapp-kubernetes-go/go.mod [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module tmp
+
+go 1.17
```
Comment (lblackstone) [inline webapp-kubernetes-go/go.mod:3]: Go 1.17 is [EOL](https://endoflife.date/go). Probably want to use 1.18

File: webapp-kubernetes-typescript/index.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+// Create a new namespace
+const webServerNs = new kubernetes.core.v1.Namespace("webserver", {metadata: {
+    name: k8sNamespace,
+}});
+
+// Create a new ConfigMap for the Nginx configuration
+const webServerConfig = new kubernetes.core.v1.ConfigMap("webserverconfig", {
+    metadata: {
+        namespace: webServerNs.metadata.apply(m => m.name),
```
Comment (lblackstone) [inline webapp-kubernetes-typescript/index.ts:20]:
```suggestion
        namespace: webServerNs.metadata.name,
```

File: webapp-kubernetes-typescript/index.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  }
+}
+`,
+    },
+});
+
+// Create a new Deployment with a user-specified number of replicas
+const webServerDeployment = new kubernetes.apps.v1.Deployment("webserverdeployment", {
+    metadata: {
+        namespace: webServerNs.metadata.apply(m => m.name),
```
Comment (lblackstone) [inline webapp-kubernetes-typescript/index.ts:42]:
```suggestion
        namespace: webServerNs.metadata.name,
```
The TS SDK will automatically lift Output values, so apply isn't needed here.

File: webapp-kubernetes-typescript/index.ts [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+                        subPath: "nginx.conf",
+                    }],
+                }],
+                volumes: [{
+                    configMap: {
+                        items: [{
+                            key: "nginx.conf",
+                            path: "nginx.conf",
+                        }],
+                        name: webServerConfig.metadata.apply(m => m.name),
```
Comment (lblackstone) [inline webapp-kubernetes-typescript/index.ts:70]:
```suggestion
                        name: webServerConfig.metadata.name,
```

File: webapp-kubernetes-typescript/index.ts [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+                }],
+            },
+        },
+    },
+});
+
+// Expose the Deployment as a Kubernetes Service
+const webServerService = new kubernetes.core.v1.Service("webserverservice", {
+    metadata: {
+        namespace: webServerNs.metadata.apply(m => m.name),
```
Comment (lblackstone) [inline webapp-kubernetes-typescript/index.ts:82]:
```suggestion
        namespace: webServerNs.metadata.name,
```

File: webapp-kubernetes-typescript/index.ts [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+            port: 80,
+            targetPort: 80,
+            protocol: "TCP",
+        }],
+        selector: appLabels,
+    },
+});
+
+// Export some values for use elsewhere
+export const deploymentName = webServerDeployment.metadata.apply(m => m.name);
```
Comment (lblackstone) [inline webapp-kubernetes-typescript/index.ts:95]:
```suggestion
export const deploymentName = webServerDeployment.metadata.name;
```

File: webapp-kubernetes-typescript/index.ts [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+export const serviceName = webServerService.metadata.apply(m => m.name);
```
Comment (lblackstone) [inline webapp-kubernetes-typescript/index.ts:96]:
```suggestion
export const serviceName = webServerService.metadata.name;
```

File: webapp-kubernetes-go/go.mod (hunk 1 above)
Comment (scottslowe) [reply webapp-kubernetes-go/go.mod]: Fixed in 0a8a73a

File: webapp-kubernetes-go/main.go [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+			return err
+		}
+
+		// Create a new ConfigMap for the Nginx configuration
+		webServerConfig, err := corev1.NewConfigMap(ctx, "webserverconfig", &corev1.ConfigMapArgs{
+			Metadata: &metav1.ObjectMetaArgs{
+				Namespace: webServerNs.Metadata.Name(),
+			},
+			Data: pulumi.StringMap{
+				"nginx.conf": pulumi.String(fmt.Sprintf(`events { }
```
[the remark this answers is not in the document]
Comment (scottslowe) [reply webapp-kubernetes-go/main.go]: Fixed in 0a8a73a

File: webapp-kubernetes-typescript/index.ts (hunk 2 above)
Comment (scottslowe) [reply webapp-kubernetes-typescript/index.ts]: Fixed in 0a8a73a

File: webapp-kubernetes-typescript/index.ts (hunk 3 above)
Comment (scottslowe) [reply webapp-kubernetes-typescript/index.ts]: Fixed in 0a8a73a

File: webapp-kubernetes-typescript/index.ts (hunk 4 above)
Comment (scottslowe) [reply webapp-kubernetes-typescript/index.ts]: Fixed in 0a8a73a

File: webapp-kubernetes-typescript/index.ts (hunk 5 above)
Comment (scottslowe) [reply webapp-kubernetes-typescript/index.ts]: Fixed in 0a8a73a

File: webapp-kubernetes-typescript/index.ts (hunk 6 above)
Comment (scottslowe) [reply webapp-kubernetes-typescript/index.ts]: Fixed in 0a8a73a

File: webapp-kubernetes-typescript/index.ts (hunk 7 above)
Comment (scottslowe) [reply webapp-kubernetes-typescript/index.ts]: Fixed in 0a8a73a

**Review by lblackstone — APPROVED** [review]
🎉

**Pull request merged**